patches/server/1052-add-number-format-api.patch

```diff
@@ -0,0 +1,175 @@
++                .getOrThrow(net.minecraft.network.chat.Style.Serializer.CODEC
++                    .parse(com.mojang.serialization.JsonOps.INSTANCE, json), com.google.gson.JsonParseException::new);
++
++            vanilla = new net.minecraft.network.chat.numbers.StyledFormat(vanillaStyle);
++        } else if (format instanceof io.papermc.paper.scoreboard.numbers.FixedFormat fixed) {
++            vanilla = new net.minecraft.network.chat.numbers.FixedFormat(io.papermc.paper.adventure.PaperAdventure
++                .asVanilla(fixed.component()));
++        } else if(format instanceof io.papermc.paper.scoreboard.numbers.BlankFormat) {
++            vanilla = net.minecraft.network.chat.numbers.BlankFormat.INSTANCE;
++        } else {
++            throw new IllegalArgumentException("Unknown format type " + format.getClass());
++        }
++
++        return vanilla;
++    }
++
++    public static NumberFormat asPaper(net.minecraft.network.chat.numbers.NumberFormat vanilla) {
++        if (vanilla instanceof net.minecraft.network.chat.numbers.StyledFormat styled) {
++            com.google.gson.JsonElement json = net.minecraft.Util
++                .getOrThrow(
++                    net.minecraft.network.chat.Style.Serializer.CODEC
++                        .encodeStart(com.mojang.serialization.JsonOps.INSTANCE, styled.getStyle()),
++                    com.google.gson.JsonParseException::new
++                );
++
++            net.kyori.adventure.text.format.Style adventureStyle =
++                net.kyori.adventure.text.serializer.gson.GsonComponentSerializer.gson()
++                    .serializer().fromJson(json, net.kyori.adventure.text.format.Style.class);
++
++            return new io.papermc.paper.scoreboard.numbers.StyledFormat(adventureStyle);
++        } else if (vanilla instanceof net.minecraft.network.chat.numbers.FixedFormat fixed) {
++            return new io.papermc.paper.scoreboard.numbers.FixedFormat(io.papermc.paper.adventure.PaperAdventure.asAdventure(fixed.getValue()));
++        } else if(vanilla instanceof net.minecraft.network.chat.numbers.BlankFormat) {
++            return io.papermc.paper.scoreboard.numbers.BlankFormat.blank();
++        }
++
++        throw new IllegalArgumentException("Unknown format type " + vanilla.getClass());
++    }
++}
+diff --git a/src/main/java/net/minecraft/network/chat/numbers/FixedFormat.java b/src/main/java/net/minecraft/network/chat/numbers/FixedFormat.java
+index 2b9f393062444a663dcd25c78f9471beda65ccf0..eb0c12016a5ef1221a3d312b71c160f03124acaf 100644
+--- a/src/main/java/net/minecraft/network/chat/numbers/FixedFormat.java
++++ b/src/main/java/net/minecraft/network/chat/numbers/FixedFormat.java
+@@ -5,6 +5,7 @@ import net.minecraft.network.FriendlyByteBuf;
+ import net.minecraft.network.chat.Component;
+ import net.minecraft.network.chat.ComponentSerialization;
+ import net.minecraft.network.chat.MutableComponent;
++import net.minecraft.network.chat.Style;
```

Comment:
Don't add this import, just use the fully qualified name inline where needed.

Reply:
That wasn't intentional and probably caused by my IDE 😅